### DISCUSSION TITLE
Discovery (.well-known)

### DISCUSSION THREAD

**Discussion by CatherineCaronCSRT** [oidc-wp/openid-connect-generic#465] [General]
Hi, could there be advantages to implement url discovery in the future (get token, authorization, userinfo endpoint urls from a .well-known endpoint instead of in the settings) ?

**Comment by timnolte** [comment]
Yes, I do think it would be helpful to try and implement this in order to ease some of the setup process. I think the big question will be whether people feel this should this be a 1 time configuration step or something that is always dynamic and can update "automatically" if the IDP configuration changes. I think a first step would be to provide an option to supply the `.well-known` endpoint to initiate a pre-population of the configuration for confirmation and saving.
